app/server.go

```diff
@@ -1454,6 +1463,90 @@ func doCheckWarnMetricStatus(a *App) {
 	}
 }
 
+func (s *Server) StopMetricsServer() {
+	s.metricsLock.Lock()
+	defer s.metricsLock.Unlock()
+
+	if s.metricsServer != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), TimeToWaitForConnectionsToCloseOnServerShutdown)
+		defer cancel()
+
+		s.metricsServer.Shutdown(ctx)
+		s.Log.Info("Metrics and profiling server is stopping", mlog.String("address", *s.Config().MetricsSettings.ListenAddress))
+		s.metricsServer = nil
+	}
+}
+
+func (s *Server) HandleMetrics(route string, h http.Handler) {
+	s.metricsRouter.Handle(route, h)
+}
+
+func (s *Server) InitMetricsRouter() {
+	s.metricsRouter = mux.NewRouter()
+	runtime.SetBlockProfileRate(*s.Config().MetricsSettings.BlockProfileRate)
+
+	rootHandler := func(w http.ResponseWriter, r *http.Request) {
+		html := `
+			<html>
+				<body>
+					<div><a href="/metrics">Metrics</a></div>
```

Comment:
nit: should this be conditionally shown?

Reply:
+1 to conditionally show if accessible. In TE, it's shown then throw 404 on click/visit.